rfc1035label/label.go

```diff
@@ -100,6 +100,14 @@ func labelsFromBytes(buf []byte) ([]string, error) {
 
 	for {
 		if pos >= len(buf) {
+			// interpret label without trailing zero-length byte as a partial
+			// domain name field as per RFC 4704 Section 4.2
+			if label != "" {
+				labels = append(labels, label)
+
+				return labels, nil
```

Comment:
Is this return needed here?

Reply:
Looking at it now it should work just as well without it. I can't run the tests right no because I'm on the phone but if they pass without the return we can get rid of it.